communication/mavlink_waypoint_handler.cpp

```diff
@@ -1456,24 +1450,20 @@ static waypoint_local_struct_t waypoint_handler_set_waypoint_from_frame(Mavlink_
 {
     global_position_t waypoint_global;
     waypoint_local_struct_t wpt;
-    local_position_t waypoint_coor;
     global_position_t origin_relative_alt;
 
     for (uint8_t i = 0; i < 3; i++)
     {
-        waypoint_coor.pos[i] = 0.0f;
+        wpt.position[i] = 0.0f;
     }
-    waypoint_coor.origin = origin;
-    waypoint_coor.heading = maths_deg_to_rad(current_waypoint->param4);
 
     switch (current_waypoint->frame)
     {
         case MAV_FRAME_GLOBAL:
             waypoint_global.latitude    = current_waypoint->x;
             waypoint_global.longitude   = current_waypoint->y;
             waypoint_global.altitude    = current_waypoint->z;
-            waypoint_global.heading     = maths_deg_to_rad(current_waypoint->param4);
-            waypoint_coor = coord_conventions_global_to_local_position(waypoint_global, origin);
+            coord_conventions_global_to_local_position(waypoint_global, origin, wpt.position);
```

Comment:
since you modify coord_conventions_global_to_local_position() shouldn't we use pointer instead ?